src/useFish.ts

```diff
@@ -13,176 +13,124 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/* eslint-disable @typescript-eslint/no-explicit-any */
-import { FishName, FishTypeImpl } from '@actyx/pond'
+import { Fish, StateEffect, Pond } from '@actyx/pond'
 import * as React from 'react'
-import { Observable } from 'rxjs'
-import { combineLatest } from 'rxjs/observable/combineLatest'
 import { usePond } from './Pond'
-import { map, switchMap, share } from 'rxjs/operators'
-
-/**
- * Feed fish function type
- *
- * @param command feed the fish with this command
- * @typeParam Command Available commands for the fish
- * @returns Promise when the feed is done
- */
-export type FeedFish<Command> = (command: Command) => Promise<void>
-/**
- * Set new fish name type. The fish name is used to hydrate the fish and return the state
- *
- * @param newName set a new name to the observing fish
- */
-export type SetNewFishName = (newName: string) => void
-/** Generic fish type for the registry fish */
-export type FishTypeRegistry = FishTypeImpl<any, any, any, ReadonlyArray<FishName | string>>
-/**
- * Generic fish type for the registry fish with generic public state
- * @typeParam PublicState Observable state of the registry fish
- */
-export type FishTypeRegistryMap<PublicState> = FishTypeImpl<any, any, any, PublicState>
-/**
- * Simplified type for the entity fish
- * @typeParam Command Available commands for the fish
- * @typeParam PublicState Observable state of the fish
- */
-export type FishTypeEntity<Command, PublicState> = FishTypeImpl<any, Command, any, PublicState>
 
 /**
  * ReactFish type.
  *
  * This fish will handle all interactions with the React component.
- *
- * state, feed, name, stream$
- *
- * @typeParam Command Available commands for the fish
- * @typeParam PublicState Observable state of the fish
  */
-export type ReactFish<Command, PublicState> = {
+export type ReactFish<State, Event, T> = {
   /** current public state of the observed fish */
-  state: PublicState
-  /** feed function for the observed fish */
-  feed: FeedFish<Command>
-  /** name of the observed fish */
-  name: string
-  /** stream$ of the public state @see useStream*/
-  stream$: Observable<PublicState>
+  state: State
+  /** run function for the observed fish */
+  run: (eff: StateEffect<State, Event>) => Promise<void>
+  /** current props of the fish */
+  props?: T
```

Comment:
I think this should also go away — React is managing props, we should stay out of its way